src/tests/cwrapper/test_cwrapper.c

```diff
@@ -0,0 +1,40 @@
+#include <stdio.h>
+#include "cwrapper.h"
+
+void test_cwrapper() {
+    char* s;
+    basic x = basic_new(), y = basic_new(), z = basic_new();
+    symbol_set(x, "x");
+    symbol_set(y, "y");
+    symbol_set(z, "z");
+    
+    s = basic_str(x);
+    printf("Symbol : %s\n", s);
+    basic e = basic_new();
+
+    integer_set_ui(e, 123);
+    s = basic_str(e);
+    printf("Integer : %s\n", s);
+
+    integer_set_ui(e, 456);
+    basic_add(e, e, x);
+    basic_mul(e, e, y);
+    basic_div(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+    basic_diff(e, e, z);
+    printf("Basic : %s\n", basic_str(e));
+
+    rational_set_ui(e, 100, 47);
+    s = basic_str(e);
+
+    printf("Rational : %s\n", s);
+    printf("Is_a_Symbol %s: %d\n", s, is_a_Symbol(e));
+    printf("Is_a_Rational %s: %d\n", s, is_a_Rational(e));
+    printf("Is_a_Integer %s: %d\n", s, is_a_Integer(e));
```

Comment:
After commenting out all str handling:
```diff
diff --git a/src/tests/cwrapper/test_cwrapper.c b/src/tests/cwrapper/test_cwrapp
index ec02247..bb19160 100644
--- a/src/tests/cwrapper/test_cwrapper.c
+++ b/src/tests/cwrapper/test_cwrapper.c
@@ -8,29 +8,34 @@ void test_cwrapper() {
     symbol_set(y, "y");
     symbol_set(z, "z");
     
-    s = basic_str(x);
-    printf("Symbol : %s\n", s);
+    //s = basic_str(x);
+    //printf("Symbol : %s\n", s);
     basic e = basic_new();
 
     integer_set_ui(e, 123);
-    s = basic_str(e);
-    printf("Integer : %s\n", s);
+    //s = basic_str(e);
+    //printf("Integer : %s\n", s);
 
     integer_set_ui(e, 456);
     basic_add(e, e, x);
     basic_mul(e, e, y);
     basic_div(e, e, z);
-    printf("Basic : %s\n", basic_str(e));
+    //printf("Basic : %s\n", basic_str(e));
     basic_diff(e, e, z);
-    printf("Basic : %s\n", basic_str(e));
+    //printf("Basic : %s\n", basic_str(e));
 
     rational_set_ui(e, 100, 47);
-    s = basic_str(e);
+    //s = basic_str(e);
 
-    printf("Rational : %s\n", s);
-    printf("Is_a_Symbol %s: %d\n", s, is_a_Symbol(e));
-    printf("Is_a_Rational %s: %d\n", s, is_a_Rational(e));
-    printf("Is_a_Integer %s: %d\n", s, is_a_Integer(e));
+    //printf("Rational : %s\n", s);
+    //printf("Is_a_Symbol %s: %d\n", s, is_a_Symbol(e));
+    //printf("Is_a_Rational %s: %d\n", s, is_a_Rational(e));
+    //printf("Is_a_Integer %s: %d\n", s, is_a_Integer(e));
+
+    basic_free(x);
+    basic_free(y);
+    basic_free(z);
+    basic_free(e);
 }
 
 int main(int argc, char* argv[])
```
It improves, but there are still some left:
```
certik@redhawk:~/repos/csympy(pr-357)$ valgrind --leak-check=full ./src/tests/cwrapper/test_cwrapper 
==28987== Memcheck, a memory error detector
==28987== Copyright (C) 2002-2012, and GNU GPL'd, by Julian Seward et al.
==28987== Using Valgrind-3.8.1 and LibVEX; rerun with -h for copyright info
==28987== Command: ./src/tests/cwrapper/test_cwrapper
==28987== 
==28987== 
==28987== HEAP SUMMARY:
==28987==     in use at exit: 32 bytes in 4 blocks
==28987==   total heap usage: 2,665 allocs, 2,661 frees, 48,523 bytes allocated
==28987== 
==28987== 8 bytes in 1 blocks are definitely lost in loss record 1 of 4
==28987==    at 0x4C285FC: operator new(unsigned long) (vg_replace_malloc.c:298)
==28987==    by 0x490CAE: basic_new (cwrapper.cpp:33)
==28987==    by 0x490B39: test_cwrapper (test_cwrapper.c:6)
==28987==    by 0x490C79: main (test_cwrapper.c:43)
==28987== 
==28987== 8 bytes in 1 blocks are definitely lost in loss record 2 of 4
==28987==    at 0x4C285FC: operator new(unsigned long) (vg_replace_malloc.c:298)
==28987==    by 0x490CAE: basic_new (cwrapper.cpp:33)
==28987==    by 0x490B47: test_cwrapper (test_cwrapper.c:6)
==28987==    by 0x490C79: main (test_cwrapper.c:43)
==28987== 
==28987== 8 bytes in 1 blocks are definitely lost in loss record 3 of 4
==28987==    at 0x4C285FC: operator new(unsigned long) (vg_replace_malloc.c:298)
==28987==    by 0x490CAE: basic_new (cwrapper.cpp:33)
==28987==    by 0x490B55: test_cwrapper (test_cwrapper.c:6)
==28987==    by 0x490C79: main (test_cwrapper.c:43)
==28987== 
==28987== 8 bytes in 1 blocks are definitely lost in loss record 4 of 4
==28987==    at 0x4C285FC: operator new(unsigned long) (vg_replace_malloc.c:298)
==28987==    by 0x490CAE: basic_new (cwrapper.cpp:33)
==28987==    by 0x490B96: test_cwrapper (test_cwrapper.c:13)
==28987==    by 0x490C79: main (test_cwrapper.c:43)
==28987== 
==28987== LEAK SUMMARY:
==28987==    definitely lost: 32 bytes in 4 blocks
==28987==    indirectly lost: 0 bytes in 0 blocks
==28987==      possibly lost: 0 bytes in 0 blocks
==28987==    still reachable: 0 bytes in 0 blocks
==28987==         suppressed: 0 bytes in 0 blocks
==28987== 
==28987== For counts of detected and suppressed errors, rerun with: -v
==28987== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: 6 from 6)
```